nscplugin/src/main/scala/scala/scalanative/nscplugin/NirGenExpr.scala

```diff
@@ -1277,7 +1279,9 @@ trait NirGenExpr { self: NirGenPhase =>
                               _))))))),
             _),
             _) =>
-          val chars = Val.Chars(str.replace("\\n", "\n").replace("\\r", "\r"))
+
+          val processedString = StringContext.processEscapes(processHexValues(str))
```

Comment:
Doing one then the other will be problematic for strings like `c"\\x65"`, which should be processed as `\x65`. But if you first call `processHexValues` you get `\A` before `processEscapes` can realize that the `\\` was an escape for `\` rather than a hex escape. Doing `processEscapes` first would cause similar issues.

I'm afraid there is no other way than dealing with everything at once.